qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/RecoveryIT.java

```diff
@@ -294,7 +295,7 @@ public void testRecoveryWithSoftDeletes() throws Exception {
                 .put(INDEX_DELAYED_NODE_LEFT_TIMEOUT_SETTING.getKey(), "100ms")
                 .put(SETTING_ALLOCATION_MAX_RETRY.getKey(), "0"); // fail faster
             if (randomBoolean()) {
-                settings.put(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), true);
+                settings.put(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), randomBoolean());
```

Comment:
why change this? Should the test be named differently in that case (e.g. just `testRecovery`)?

Reply:
We introduced this test when soft-deletes was disabled by default. We should invert the test condition as soft-deletes is enabled by default now. I renamed it in https://github.com/elastic/elasticsearch/pull/50502/commits/893d28ea7ad030c29390160163f952e4ea421a67.